File: ios/ShortDrama/Sources/Domain/UseCases/GetCurrentUserUseCase.swift
import Foundation

struct GetCurrentUserUseCase: Sendable {
    private let repository: AuthRepositoryProtocol

    init(repository: AuthRepositoryProtocol) {
        self.repository = repository
    }

    func execute(accessToken: String) async throws -> AuthUser {
        try await repository.getCurrentUser(accessToken: accessToken)
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/usecase/GetCurrentUserUseCase.kt
package com.djs66256.short_drama.domain.usecase

import com.djs66256.short_drama.core.network.ApiResult
import com.djs66256.short_drama.domain.model.AuthUser
import com.djs66256.short_drama.domain.repository.AuthRepository
import javax.inject.Inject

class GetCurrentUserUseCase @Inject constructor(
    private val authRepository: AuthRepository,
) {
    suspend operator fun invoke(): ApiResult<AuthUser> {
        return authRepository.getCurrentUser()
    }
}
